### ISSUE TITLE
[Bug Report] invalid hardsigmoid backward result

### ISSUE THREAD

**Issue by hschoi4448** [tenstorrent-metal/tt-metal#6534]
**Describe the bug**

The `hardsigmoid_bw` function returns an invalid gradient value.

**To Reproduce**
1. Copy and past below code to `/tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_hardsigmoid.py `
```Python
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

import torch
import pytest
import tt_lib
from tests.tt_eager.python_api_testing.unit_testing.backward_ops.utility_funcs import compare_results, data_gen_pt_tt


def data_gen_pt_tt(input_shapes, device, required_grad=False, val=1):
    pt_tensor = (torch.ones(input_shapes, requires_grad=required_grad) * val).bfloat16()
    tt_tensor = (
        tt_lib.tensor.Tensor(pt_tensor, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
    )
    return pt_tensor, tt_tensor


@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
def test_bw_hardsigmoid(input_shapes, device):
    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True, val=10)
    grad_data, grad_tensor = data_gen_pt_tt(input_shapes, device, False, 100)
    
    print("input_tensor", input_tensor) # 10
    print("grad_tensor", grad_tensor) # 100

    pyt_y = torch.nn.functional.hardsigmoid(in_data)

    tt_output_tensor_on_device = tt_lib.tensor.hardsigmoid_bw(grad_tensor, input_tensor)

    in_data.retain_grad()

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]
    comp_pass = compare_results(tt_output_tensor_on_device, golden_tensor)

    print("tt_output_tensor_on_device", tt_output_tensor_on_device) # 16.62500
    print("golden_tensor", golden_tensor) # 0
    assert comp_pass # Assert False here
```
2. Run cmd `pytest ./tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_hardsigmoid.py
```Python
input_tensor ttnn.Tensor([[[[10.00000, 10.00000,  ..., 10.00000, 10.00000],
               [10.00000, 10.00000,  ..., 10.00000, 10.00000],
               ...,
               [10.00000, 10.00000,  ..., 10.00000, 10.00000],
               [10.00000, 10.00000,  ..., 10.00000, 10.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
grad_tensor ttnn.Tensor([[[[100.00000, 100.00000,  ..., 100.00000, 100.00000],
               [100.00000, 100.00000,  ..., 100.00000, 100.00000],
               ...,
               [100.00000, 100.00000,  ..., 100.00000, 100.00000],
               [100.00000, 100.00000,  ..., 100.00000, 100.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
tt_output_tensor_on_device [ttnn.Tensor([[[[16.62500, 16.62500,  ..., 16.62500, 16.62500],
               [16.62500, 16.62500,  ..., 16.62500, 16.62500],
               ...,
               [16.62500, 16.62500,  ..., 16.62500, 16.62500],
               [16.62500, 16.62500,  ..., 16.62500, 16.62500]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)]
golden_tensor [tensor([[[[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]]]], dtype=torch.bfloat16)]
```

**Expected behavior**

I want `hardsigmoid_bw` to return the correct gradient

**Screenshots**

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit e14b5748466586a4d139ea7179feedff1886d474)

**Additional context**

**Comment by umadevimcw** [comment]
Merged to main
